File: fizzbuzz/src/zig/fizzbuzz.zig
const std = @import("std");
const stdout = std.io.getStdOut().writer();

pub fn main() !void {
    var n: i64 = 1;
    while (n <= 50) {
        if (@mod(n, 15) == 0) {
            try stdout.print("fizz buzz\n", .{});
        } else if (@mod(n, 3) == 0) {
            try stdout.print("fizz\n", .{});
        } else if (@mod(n, 5) == 0) {
            try stdout.print("buzz\n", .{});
        } else {
            try stdout.print("{}\n", .{n});
        }
        n += 1;
    }
}

File: fizzbuzz/src/ocaml/fizzbuzz.ml
for n = 1 to 50 do
  if n mod 3 == 0 && n mod 5 == 0
    then Printf.printf "fizz buzz\n"
  else if n mod 3 == 0
    then Printf.printf "fizz\n"
  else if n mod 5 == 0
    then Printf.printf "buzz\n"
  else Printf.printf "%d\n" n
done;;
